### ISSUE TITLE
Windows build script has side effects on the PATH

### ISSUE THREAD

**Issue by maximedenes** [coq/coq#7334]
The Windows build script changes `.bash_profile` in a way that persists after a build. I think we should minimize such effects, so that there is no difference between a build in a clean environment and a build in the same environment as a previous one.

I don't fully understand what `configure_profile.sh` does. Isn't there a way to set up the path locally to a shell without having to modify `.bash_profile`?

@MSoegtropIMC probably knows more.

**Comment by MSoegtropIMC** [comment]
The scripts have originally been designed to also setup a working environment to work with Coq on Windows. Since a dedicated Cygwin is setup, it makes sense to adjust the .bashrc in this dedicated Cygwin in a sensible way (I explain below what I mean by this). Originally it was so that on CI also each time a new Cygwin was setup, but I think this changed some time last year for speed reasons (not 100% sure). In case Cygwins are reused, I agree that this is suboptimal. But reusing Cygwin setups (without extensive checks) is also suboptimal and in part defies the intention of the scripts to setup a clean known environment. I will check how this works and in case create an issue to add Cygwin configuration checks.

Regarding the question what configure_profile.sh does:

1.) It creates proxy settings. The proxy settings are inherited from the HTTP_PROXY environment variable in MakeCoq_MinGW.bat (unless given on the command line). Without this it is not possible to download files if the machine is behind a proxy.

2.) It shortens the PATH environment variable. The PATH environment variable can be so long that some utilities (especially xargs and find) exit with an error message that the environment is too large. Also an uncontrolled PATH can lead to unreproducible results. ORIGINAL_PATH (an environment variable set by Cygwin) is also cleared for this reason.

3.) The OCAMLLIB environment variable is cleared to avoid interference with existing OCaml installations. OCAMMLIB overrides the built in path of OCaml which is not what one wants here.

In summary I think that these changes always make sense, but of cause it wouldn't be a problem to make them locally in makecoq_mingw.sh. What won't work then any more is to continue a failed build with calling make from a Cygwin shell (that is not via makecoq_mingw.sh). Many of the makefiles will fail without these changes (at least on most personal computers with a longer PATH) since xargs and find will always exit. So I would suggest to add a -CI switch to MakeCoq_MinGW.bat to choose if this should be done locally in makecoq_mingw.sh or globally in .bashrc.

Best regards,

Michael

**Comment by maximedenes** [comment]
@MSoegtropIMC That makes sense, thanks. So maybe the best option is to make sure our CI starts with a fresh Cygwin each time. I can do that, so I'm not sure working on the extra check is necessary.

**Comment by MSoegtropIMC** [comment]
@maximedenes I think (not sure) that reinstalling cgwin each time was discarded some time last year cause of speed reasons. When Cygwin is installed each time the probability of hitting the 1 hour limit is much higher. Of cause this means that it still has a good probability of failing when the build finds a fresh machine, but on average not installing Cygwin each time seems to be the better solution.

Let me review how the "is Cygwin available" check is done and if it can be changes such that e.g. path changes are ok. I guess with a -CI option one can come to a reasonable solution. For non CI usage I would leave it as is. Otherwise I expect a lot of questions about these failures on the list (well it would give me feedback on how many people are using this ;-)

**Comment by MSoegtropIMC** [comment]
This is deprecated with Coq Platform.
